sipi/scripts/get_knora_session.lua

```diff
@@ -17,13 +19,37 @@ function get_session_id(cookie)
         return nil
     end
 
+    -- name of the cokie depends on the environment defined as host:port combination
+    -- this combination is "mangled" using base32 and appended to "KnoraAuthentication"
+    -- to get the correct cokie, we need to calculate first the mangled host-port combination
+    local webapi_hostname = os.getenv("KNORA_WEBAPI_KNORA_API_EXTERNAL_HOST")
+    local webapi_port = os.getenv("KNORA_WEBAPI_KNORA_API_EXTERNAL_PORT")
+    if webapi_hostname == nil then
+        send_error(500, "KNORA_WEBAPI_KNORA_API_EXTERNAL_HOST not set")
+        return nil
+    end
+    if webapi_port == nil then
+        send_error(500, "KNORA_WEBAPI_KNORA_API_EXTERNAL_PORT not set")
+        return nil
+    end
+    
+    host_port = webapi_hostname .. ':' .. webapi_port
+    server.log("host_port: " .. host_port, server.loglevel.LOG_DEBUG)
+
+    local customPadMap = { "", "999999", "9999", "999", "9" }
```

Comment:
just out of curiosity: what's the deal with this padMap?

Reply:
the standard padding is `=` which is not allowed because then the heather is going to be something like `KnoraAuthenticationDFJSKLFJDSLJ===JWT`. No additional equal signs are allowed. This is why I changed it to `9` which is not used as part of the alphabet in the algorithm.